.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,152 @@
+
+
+### Debug ZFS
+
+A variety of methods and tools are available to aid ZFS developers.  It's strongly recommended
+that when developing a patch the `--enable-debug` configure option should be set.
+This will enable additional correctness checks and all the ASSERTs to help quickly catch potential issues.
+
+In addition, there are numerous utilities and debugging files which provide visibility in to
+the inner workings of ZFS.  The most useful of these tools are discussed in detail on the
+[debugging ZFS wiki page](https://github.com/zfsonlinux/zfs/wiki/Debugging).
+
+
+### Where can I ask for help?
+The [mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists) is the best place to ask for help.
+
+Please *note*, our issue tracker is **only for bugs**. 
+If you have any comments or suggestions regarding any enhancements, please use our 
+[mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists).
+
+
+## How Can I Contribute?
+
+
+### Reporting Bugs
+
+*Please* contact us via the [mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists) if you aren't certain that you are experiencing a bug.
+
+If you run into an issue, please search our [issue tracker](https://github.com/zfsonlinux/zfs/issues) *first* to ensure the issue hasn't been reported before. 
+Open a new issue only if you haven't found anything similar to your issue.
+
+You can open a new issue and search existing issues using the public
+[issue tracker](https://github.com/zfsonlinux/zfs/issues). 
+
+
+#### When opening a new issue, please include the following information at the top of the issue:
+
+* What distribution (with version) you are using.
+* The spl and zfs versions you are using, installation method (repository or manual compilation).
+* Describe the issue you are experiencing.
+* Describe how to reproduce the issue.
+* Including any warning/errors/backtraces from the system logs.
+
+When a new issue is opened, it is not uncommon for developers to request additional information.
+
+In general, the more detail you share about a problem the quicker a developer 
+can resolve it. For example, providing a simple test case is always 
+exceptionally helpful.  
+
+Be prepared to work with the developers investigating your issue. 
+Your assistance is crucial in providing a quick solution.
+They may ask for information like:
+
+* Your pool configuration as reported by `zdb` or `zpool status`.
+* Your hardware configuration, such as
+  * Number of CPUs.
+  * Amount of memory.
+  * Whether your system has ECC memory.
+  * Whether it is running under a VMM/Hypervisor.
+  * Kernel version.
+  * Values of the spl/zfs module parameters.
+* Stack traces which may be logged to `dmesg`.
+
+
+### Suggesting Enhancements
+
+ZFS on Linux is a widely deployed production filesystem which is under active development.
+The team's primary focus is on fixing known issues, improving performance and adding compelling new features.
```

Comment:
Need a comma after performance. Should be `..known issues, improving performance, and...`